zebra-chain/src/transaction/unmined.rs

```diff
@@ -353,17 +356,24 @@ impl VerifiedUnminedTx {
         transaction: UnminedTx,
         miner_fee: Amount<NonNegative>,
         legacy_sigop_count: u64,
-    ) -> Self {
+    ) -> Result<Self, zip317::Error> {
         let fee_weight_ratio = zip317::conventional_fee_weight_ratio(&transaction, miner_fee);
         let unpaid_actions = zip317::unpaid_actions(&transaction, miner_fee);
 
-        Self {
+        zip317::mempool_checks(
+            unpaid_actions,
+            BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT,
+            miner_fee,
+            transaction.conventional_fee,
+        )?;
+
```

Comment:
As part of this ticket, we also wanted to enforce a minimum mempool fee, like `zcashd` does. This is not from ZIP-317, it's from Bitcoin.

Here's Daira's summary of what `zcashd` does:
https://github.com/ZcashFoundation/zebra/issues/5336#issuecomment-1506748801

I think the ticket was really complicated, there were a lot of updates over a few months. I've just edited it to remove the outdated and completed work.